src/RestServer/RestServerUtility.JTokens.cs

```diff
@@ -0,0 +1,330 @@
+                    var s = condition as ScriptHashCondition;
+                    j = JToken.FromObject(new
+                    {
+                        s.Type,
+                        s.Hash,
+                    }, serializer);
+                    break;
+                case WitnessConditionType.Group:
+                    var g = condition as GroupCondition;
+                    j = JToken.FromObject(new
+                    {
+                        g.Type,
+                        g.Group,
+                    }, serializer);
+                    break;
+                case WitnessConditionType.CalledByEntry:
+                    var e = condition as CalledByEntryCondition;
+                    j = JToken.FromObject(new
+                    {
+                        e.Type,
+                    }, serializer);
+                    break;
+                case WitnessConditionType.CalledByContract:
+                    var c = condition as CalledByContractCondition;
+                    j = JToken.FromObject(new
+                    {
+                        c.Type,
+                        c.Hash,
+                    }, serializer);
+                    break;
+                case WitnessConditionType.CalledByGroup:
+                    var p = condition as CalledByGroupCondition;
+                    j = JToken.FromObject(new
+                    {
+                        p.Type,
+                        p.Group,
+                    }, serializer);
+                    break;
+                default:
+                    break;
+            }
+            return j;
+        }
+
+        public static JToken ContractStateToJToken(ContractState contract, global::Newtonsoft.Json.JsonSerializer serializer) =>
+            JToken.FromObject(new
+            {
+                contract.Id,
+                contract.Manifest.Name,
+                contract.Hash,
+                Manifest = ContractManifestToJToken(contract.Manifest, serializer),
+                NefFile = ContractNefFileToJToken(contract.Nef, serializer),
+            }, serializer);
+
+        public static JToken ContractManifestToJToken(ContractManifest manifest, global::Newtonsoft.Json.JsonSerializer serializer) =>
+            JToken.FromObject(new
+            {
+                manifest.Name,
+                Abi = ContractAbiToJToken(manifest.Abi, serializer),
+                Groups = manifest.Groups.Select(s => ContractGroupToJToken(s, serializer)),
+                Permissions = manifest.Permissions.Select(s => ContractPermissionToJToken(s, serializer)),
+                Trusts = manifest.Trusts.Select(s => ContractPermissionDescriptorToJToken(s, serializer)),
+                manifest.SupportedStandards,
+                Extra = manifest.Extra.Count > 0 ?
+                    new JObject(manifest.Extra.Properties.Select(s => new JProperty(s.Key.ToString(), s.Value.AsString()))) :
+                    null,
```

Comment:
NullReferenceException will be threw here when getting native contract manifest

Reply:
Which line?